app/models/user.rb

```diff
@@ -456,6 +456,16 @@ def close_account!
     AccountDeletion.create(:person => self.person)
   end
 
+  def lock_account!
+    self.locked_at = Time.now
+    self.save
+  end
```

Comment:
Proof that we're already using that functionality: https://github.com/diaspora/diaspora/blob/develop/app/models/user.rb#L479